src/PyEventLoop.cc

```diff
@@ -28,21 +28,37 @@ static PyObject *eventLoopJobWrapper(PyObject *jobFn, PyObject *Py_UNUSED(_)) {
 }
 static PyMethodDef loopJobWrapperDef = {"eventLoopJobWrapper", eventLoopJobWrapper, METH_NOARGS, NULL};
 
+static PyObject *_enqueueWithDelay(PyObject *_loop, PyEventLoop::AsyncHandle::id_t handleId, PyObject *jobFn, double delaySeconds, bool repeat);
+
 /**
  * @brief Wrapper to remove the reference of the timer after the job finishes
  */
-static PyObject *timerJobWrapper(PyObject *jobFn, PyObject *handlerPtr) {
-  auto handle = (PyEventLoop::AsyncHandle *)PyLong_AsVoidPtr(handlerPtr);
+static PyObject *timerJobWrapper(PyObject *jobFn, PyObject *args) {
+  PyObject *_loop = PyTuple_GetItem(args, 0);
+  PyEventLoop::AsyncHandle::id_t handleId = PyLong_AsLong(PyTuple_GetItem(args, 1));
+  double delaySeconds = PyFloat_AsDouble(PyTuple_GetItem(args, 2));
+  bool repeat = (bool)PyLong_AsLong(PyTuple_GetItem(args, 3));
+  auto handle = PyEventLoop::AsyncHandle::fromId(handleId);
+
   PyObject *ret = PyObject_CallObject(jobFn, NULL); // jobFn()
   Py_XDECREF(ret); // don't care about its return value
-  handle->removeRef();
-  if (PyErr_Occurred()) {
+
+  PyObject *errType, *errValue, *traceback; // we can't call any Python code unless the error indicator is clear
+  PyErr_Fetch(&errType, &errValue, &traceback);
+  if (repeat && !handle->cancelled()) {
+    _enqueueWithDelay(_loop, handleId, jobFn, delaySeconds, repeat);
+  } else {
+    handle->removeRef();
+  }
+
+  if (errType != NULL) { // PyErr_Occurred()
+    PyErr_Restore(errType, errValue, traceback);
     return NULL;
   } else {
     Py_RETURN_NONE;
```

Comment:
```suggestion
    Py_XDECREF(errValue);
    Py_XDECREF(traceback);
    Py_RETURN_NONE;
```

Reply:
Not actually needed.

https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Fetch
> If the error indicator is not set, all three variables are set to NULL